include/seqan3/range/container/bitcompressed_vector.hpp

```diff
@@ -0,0 +1,1101 @@
+         * \{
+         */
+        //!\brief Implicitly convertible to alphabet_type.
+        constexpr operator alphabet_type() const
+            noexcept(noexcept(assign_rank(alphabet_type{}, static_cast<uint64_t>(internal_proxy))))
+        {
+            return assign_rank(alphabet_type{}, static_cast<uint64_t>(internal_proxy));
+        }
+
+        //!\brief Assignable from alphabet_type.
+        constexpr reference_proxy_type & operator=(alphabet_type const a)
+            noexcept(noexcept(internal_proxy = to_rank(a)))
+        {
+            internal_proxy = to_rank(a);
+            return *this;
+        }
+        //!\}
+
+        //!\name Comparison operators (self)
+        //!\{
+        constexpr bool operator==(reference_proxy_type const & rhs) const noexcept
+        {
+            return internal_proxy == rhs.internal_proxy;
+        }
+
+        constexpr bool operator!=(reference_proxy_type const & rhs) const noexcept
+        {
+            return !(internal_proxy == rhs.internal_proxy);
+        }
+
+        constexpr bool operator<(reference_proxy_type const & rhs) const noexcept
+        {
+            return internal_proxy < rhs.internal_proxy;
+        }
+
+        constexpr bool operator>(reference_proxy_type const & rhs) const noexcept
+        {
+            return !(internal_proxy <= rhs.internal_proxy);
+        }
+
+        constexpr bool operator<=(reference_proxy_type const & rhs) const noexcept
+        {
+            return internal_proxy == rhs.internal_proxy || internal_proxy < rhs.internal_proxy;
+        }
+
+        constexpr bool operator>=(reference_proxy_type const & rhs) const noexcept
+        {
+            return !(internal_proxy < rhs.internal_proxy);
+        }
+        //!\}
+
+        //!\name Comparison operators (alphabet_type)
+        //!\{
+        constexpr bool operator==(alphabet_type const & rhs) const noexcept
+        {
+            return static_cast<alphabet_type>(*this) == rhs;
+        }
+
+        constexpr bool operator!=(alphabet_type const & rhs) const noexcept
+        {
+            return !(static_cast<alphabet_type>(*this) == rhs);
+        }
+
+        constexpr bool operator<(alphabet_type const & rhs) const noexcept
+        {
+            return static_cast<alphabet_type>(*this) < rhs;
+        }
+
+        constexpr bool operator>(alphabet_type const & rhs) const noexcept
+        {
+            return !(static_cast<alphabet_type>(*this) <= rhs);
+        }
+
+        constexpr bool operator<=(alphabet_type const & rhs) const noexcept
+        {
+            return static_cast<alphabet_type>(*this) == rhs || static_cast<alphabet_type>(*this) < rhs;
+        }
+
+        constexpr bool operator>=(alphabet_type const & rhs) const noexcept
+        {
+            return !(static_cast<alphabet_type>(*this) < rhs);
+        }
+        //!\}
+
+        //NOTE(h-2): If the alphabets ever get ++ -- a.s.o. this proxy should, too.
```

Comment:
Could we enforce this with a `static_assert()`? Like 
```cpp
static_assert(!incrementable_concept<alphabet_type>, "TODO: Adapt this class to also support increment operation?")
```
So, we would be forced to not forget it.